### DISCUSSION TITLE
How to add dropdowns, dividers and order elements in the toolbar

### DISCUSSION THREAD

**Discussion by Stupidism** [payloadcms/payload#980] [Q&A]
Now my toolbar looks like this:

<img width="636" alt="image" src="https://user-images.githubusercontent.com/5801015/185799399-44d952fc-eef2-416f-8624-93e363c947f9.png">

But I want to implement sth similar with this:

<img width="732" alt="image" src="https://user-images.githubusercontent.com/5801015/185799469-d339672f-5196-4f2f-9ec7-82071ba375a3.png">

Now the leaves buttons are placed after elements buttons.

**Comment by Stupidism** [comment]
Bump it up

**Comment by jmikrut** [comment]
Hey @Stupidism — this is a good question. I think the solution here would be to extend the Rich Text field with a customizable `toolbar` component, where it would accept the built-in button components as a prop, and then it could render whatever UI necessary.

We would happily accept this as a PR as I think this would be a great addition to the core! In the meantime, this would be totally possible by just swapping in your own rich text component and pulling what you need out of Payload's built in rich text component.

Thoughts?
